server/scores.js

```diff
@@ -4,27 +4,77 @@ const express = require('express')
 const Promise = require('bluebird')
 const { MongoClient, ObjectID } = require('mongodb')
 const { authroizationMiddlware } = require('@first-lego-league/ms-auth')
+const Configuration = require('@first-lego-league/ms-configuration')
 
 const DEFAULTS = require('./defaults')
 
 const mongoUrl = process.env.MONGO_URI || DEFAULTS.MONGO
 
 const router = express.Router()
 
+class MissingFieldError extends Error {
+  constructor (err) {
+    super()
+    this.error = err
+    Error.captureStackTrace(this, MissingFieldError)
+  }
+}
+
+const ERROR = {
+  TEAM_NUMBER: 'team number ',
+  SCORE: 'score ',
+  MATCH: 'match ',
+  NONE: ''
+}
+
 const connectionPromise = MongoClient
   .connect(mongoUrl, { promiseLibrary: Promise, useNewUrlParser: true })
   .then(client => client.db().collection('scores'))
 
+function _validateScore (score) {
+  let validatedScore = score
+
+  let missingFieldError = new MissingFieldError(ERROR.NONE)
+
+  return Configuration.get('autoPublish').then(autoPublishSetting => {
+    validatedScore.published = autoPublishSetting
+
+    if (typeof validatedScore.teamNumber !== 'number') { missingFieldError.error += ERROR.TEAM_NUMBER }
+    if (validatedScore.score == null) { missingFieldError.error += ERROR.SCORE }
+    if (validatedScore.match == null) { missingFieldError.error += ERROR.MATCH }
+
+    if (missingFieldError.error !== ERROR.NONE) { throw missingFieldError }
+
+    return validatedScore
+  }).catch(err => {

```

Comment:
What is that point of this catch?

Reply:
Earlier there was an error if it did not exist, because I a returning the promise now there is no problem omitting it.